### ISSUE TITLE
ImplicitDefaultLocale does not contain kotlin 1.5 functions

### ISSUE THREAD

**Issue by zmunm** [detekt/detekt#3966]
## Expected Behavior

We expect ImplicitDefaultLocale to have consistent behavior for functions added in kotlin 1.5.

## Observed Behavior

It only works for toUpperCase and toLowerCase.

## Steps to Reproduce

```kotlin
"foo".lowercase()
```

## Context

https://kotlinlang.org/docs/whatsnew15.html#stable-locale-agnostic-api-for-upper-lowercasing-text

However, kotlin 1.5 suggests `String.replaceFirstChar { it.uppercase() }` instead of `capitalize()`.
The existing rule will report this, but in this case, I am concerned that it may be a false positive.

## Your Environment

* Version of detekt used: 1.17.1
* Version of Gradle used (if applicable): 7.1
* Gradle scan link (add `--scan` option when running the gradle task):
* Operating System and version:
* Link to your project (if it's a public repository):

**Comment by cortinico** [comment]
> It only works for toUpperCase and toLowerCase.

I believe that's actually the correct behavior.

Using `.uppercase()` and `.lowercase()` without an explicit Locale is fine as those functions will use the invariant locale. The former `toUpperCase` and `toLowerCase` would use instead `Locale.getDefault()` on the JVM (that's what the rule is trying to prevent).

**Comment by zmunm** [comment]
I completely misunderstood `Locale.ROOT` and `Locale.getDefault()` and I agree that the behavior is correct now. 😞
Thanks for sharing your insight. 👍
